.eleventy.js

```diff
@@ -30,6 +30,33 @@ const DEV_MODE = process.env.ELEVENTY_RUN_MODE !== "build" // i.e. serve/watch
 module.exports = function(eleventyConfig) {
     eleventyConfig.setUseGitIgnore(false); // Otherwise docs are ignored
 
+    // Set DEV_MODE_POSTS to true if the context is not 'production'
+    const DEV_MODE_POSTS = process.env.CONTEXT !== "production";
+
+    let hasPrintedMessage = false;
+
+    // Define a filter named 'isFuturePost'
+    let processedPosts = {};
+    eleventyConfig.addFilter('isFuturePost', (post) => {
+        const isFuturePost = post.date && post.date > new Date();
+        if (isFuturePost && !processedPosts[post.title]) {
```

Comment:
`post.title` is a weak hash IMO. Next time it should likely just be a an actual hash of the `title + date`. That combo is likely to be unique.

Further, what's interesting is you're using a hash/map data structure exactly how the `Set` is implemented. :) Next time consider using a `Set`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

Reply:
Something like this?

```
    let processedPosts = new Set();
    eleventyConfig.addFilter('isFuturePost', (post) => {
        const isFuturePost = post.date && post.date > new Date();
        const postIdentifier = `${post.title}-${post.date}`;
        if (isFuturePost && !processedPosts.has(postIdentifier)) {
            let text = DEV_MODE_POSTS ? 'Including' : 'Excluding';
            let formattedDate = eleventyConfig.getFilter('shortDate')(post.date);
            console.log(`[11ty/eleventy-base-blog] ${text} ${post.title} scheduled for ${formattedDate}`);
            processedPosts.add(postIdentifier);
        }
        return isFuturePost && !DEV_MODE_POSTS;
    });
```